mlflow/projects/kubernetes.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import
+import yaml
+import logging
+import docker
+import os
+import time
+import kubernetes
+from datetime import datetime
+
+_logger = logging.getLogger(__name__)
+
+
+def push_image_to_registry(image_uri):
+    client = docker.from_env()
+    _logger.info("=== Pushing docker image %s ===", image_uri)
+    client.images.push(repository=image_uri)
```

Comment:
It looks like this line doesn't always throw on error cases. See the following:
```
ipdb> p client.images.push(repository=image_uri)
u'{"status":"The push refers to repository [mlflow-registry.dev.databricks.com/andrew/mlflow-test]"}\r\n{"errorDetail":{"message":"Get https://mlflow-registry.dev.databricks.com/v2/: Service Unavailable"},"error":"Get https://mlflow-registry.dev.databricks.com/v2/: Service Unavailable"}\r\n'
ipdb> p type(client.images.push(repository=image_uri))
<type 'unicode'>```

Reply:
yes, this one was tricky. The error isn't an API error, but now it's checking for errors in response.

Done